crates/ruff_linter/src/rules/refurb/rules/write_whole_file.rs

```diff
@@ -54,7 +54,7 @@ impl Violation for WriteWholeFile {
 /// FURB103
 pub(crate) fn write_whole_file(checker: &mut Checker, with: &ast::StmtWith) {
     // `async` check here is more of a precaution.
-    if with.is_async || !checker.semantic().is_builtin("open") {
+    if with.is_async {
```

Comment:
I think it would be good to add additional tests that cover the newly added logic. E.g. that it now supports `builtins.open` or `from bultins import open as o` (assuming this is indeed now supported). 

Another test could be 

```
with (a as open, open(b) as b):
    ...
```

The added tests would also demonstrate that the genrated fixes work correctly.